### PR TITLE
Destination Kafka: correct spec json and data types in config

### LINKED ISSUE #5789: Kafka Destination connector Issue in UI
## Enviroment
- **Airbyte version**:  0.29.8-alpha
- **OS Version / Instance**: macOS
- **Deployment**: Docker 
- **Source Connector and version**:  NA
- **Destination Connector and version**: 2.8.0
Screenshot 2021-08-27 at 5 50 41 PM

- **Severity**:  Medium 
- **Step where error happened**: while creating Kafka Destination in UI

## Current Behavior
When Kafka destination is getting created from UI, It giving error like below 
-----------------------------------------------------------------------------------------------
Verification error(s) occurred for CHECK. Errors: [$.buffer_memory: is missing but it is required, $.max_block_ms: is missing but it is required, $.linger_ms: is missing but it is required, $.socket_connection_setup_timeout_max_ms: is missing but it is required, $.socket_connection_setup_timeout_ms: is missing but it is required]
-----------------------------------------------------------------------------------------------

when I update fields value getting below error :
---------------------------------------------------------------------------------------------------------
Verification error(s) occurred for CHECK. Errors: [$.buffer_memory: string found, number expected, $.max_block_ms: string found, number expected, $.socket_connection_setup_timeout_max_ms: 
---------------------------------------------------------------------------------------------------------

Its not able to create kafka destination due to above error.
It seems like a bug, as fields value are coming as string instead of integer and thats why its giving error.

## Expected Behavior
It should accept default value of fields
It should send fields value as integer not as string.
Kafka destination should be created properly.

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```

2021-09-01 16:09:53 INFO () TemporalAttemptExecution(get):110 - Executing worker wrapper. Airbyte version: 0.29.8-alpha
2021-09-01 16:09:53 INFO () LineGobbler(voidCall):85 - Checking if airbyte/destination-kafka:0.1.1 exists...
2021-09-01 16:09:53 INFO () LineGobbler(voidCall):85 - airbyte/destination-kafka:0.1.1 was found locally.
2021-09-01 16:09:53 INFO () DockerProcessFactory(create):146 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/5c735263-7ff1-4aa9-8f2d-819056e01958/0 --network host --log-driver none airbyte/destination-kafka:0.1.1 check --config source_config.json
2021-09-01 16:09:55 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-09-01 16:09:55 INFO i.a.i.d.k.KafkaDestination(main):104 - {} - Starting destination: class io.airbyte.integrations.destination.kafka.KafkaDestination
2021-09-01 16:09:56 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-09-01 16:09:56 INFO i.a.i.b.IntegrationRunner(run):96 - {} - Running integration: io.airbyte.integrations.destination.kafka.KafkaDestination
2021-09-01 16:09:56 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-09-01 16:09:56 INFO i.a.i.b.IntegrationCliParser(parseOptions):135 - {} - integration args: {check=null, config=source_config.json}
2021-09-01 16:09:56 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-09-01 16:09:56 INFO i.a.i.b.IntegrationRunner(run):100 - {} - Command: CHECK
2021-09-01 16:09:56 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-09-01 16:09:56 INFO i.a.i.b.IntegrationRunner(run):101 - {} - Integration config: IntegrationConfig{command=CHECK, configPath='source_config.json', catalogPath='null', statePath='null'}
2021-09-01 16:09:56 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-09-01 16:09:56 WARN c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword examples - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2021-09-01 16:09:56 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-09-01 16:09:56 WARN c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword airbyte_secret - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2021-09-01 16:09:56 INFO () TemporalAttemptExecution(get):133 - Stopping cancellation check scheduling...

```

</details>

## Steps to Reproduce
1. Deploy airbyte on Mac machine on docker 
2. create Kafka destination from UI.
3. Pu all required parameters , error will occur.

## Are you willing to submit a PR?

NO

### LINKED ISSUE #6039: Destination Kafka: config fields are not converting to correct data types
## Enviroment
- **Airbyte version**:  0.29.16-alpha
- **OS Version / Instance**: MacOS
- **Deployment**: Docker
- **Source Connector and version**: ˜
- **Destination Connector and version**: Kafka 0.1.1.
- **Severity**: Very Low / Low / Medium / High / Critical
- **Step where error happened**: Deploy / Sync job / Setup new connection / Update connector / Upgrade Airbyte

## Current Behavior
image
2 problems:
1. default values are not passed to API request
2. number field are convert to string and break config parsing function.

## Expected Behavior
*Tell us what should happen.*

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```

replace this with
your long log
output here

```

</details>

## Steps to Reproduce
1.
2.
3.

## Are you willing to submit a PR?

Remove this with your answer.

### PR DESCRIPTION
**Pull request by marcosmarxm**
## What
Closes https://github.com/airbytehq/airbyte/issues/6039

## How
Default value in spec.json are not being sent to API request => change default to examples.
number type is converted to string and has conflict, convert to string in spec.json and convert in config parse function only.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
After the change I was able to connect using the UI and create Kafka destination connector. The check function is not strong right now... need work.
image

File: airbyte-integrations/connectors/destination-kafka/src/main/resources/spec.json [hunk 1]
```diff
@@ -191,62 +191,62 @@
...
         "type": "integer",
-        "default": 131072
+        "examples": [131072]
       },
       "receive_buffer_bytes": {
         "title": "Receive buffer bytes",
         "description": "The size of the TCP receive buffer (SO_RCVBUF) to use when reading data. If the value is -1, the OS default will be used.",
         "type": "integer",
-        "default": 32768
+        "examples": [32768]
```
Comment (sherifnada) [inline airbyte-integrations/connectors/destination-kafka/src/main/resources/spec.json:249]: why not keep the default?

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/destination-kafka/src/main/resources/spec.json (hunk 1 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/destination-kafka/src/main/resources/spec.json:249]: Right now Default value in spec.json is not being sent to API request => change the default to examples and force user to insert value.

Comment (marcosmarxm) [reply airbyte-integrations/connectors/destination-kafka/src/main/resources/spec.json:249]: I'll create another issue to investigate this in the front-end.

**Pull request merged**